File: Proyecto Migrar/MainApp/ClsDataApp/CTemplate.vb
Imports System.Data
Imports System.Data.SqlClient
Namespace SIS.Data
    Public Class CTemplate
        Implements IDisposable

        Protected ObjConnection As SqlConnection
        Protected ObjAdapter As SqlDataAdapter
        Protected ObjParam As SqlParameter
        Protected Objcommand As SqlCommand
        Protected ObjReader As SqlDataReader
        ' Protected StrConnection As String = ConfigurationSettings.AppSettings.Item("StrSistConn")

        Public Sub Dispose() Implements System.IDisposable.Dispose

            Try
                If ObjConnection.State <> ConnectionState.Closed Then
                    ObjConnection.Close()
                End If

            Catch ex As Exception

            End Try

            ObjConnection = Nothing
            ObjAdapter = Nothing
            ObjParam = Nothing
            Objcommand = Nothing
            ObjReader = Nothing

        End Sub
    End Class
End Namespace

File: PruebaMenu/App-Base/AppBaseSIS/ClsDataApp/CTemplate.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SIS.Data
{
    public class CTemplate : IDisposable
    {
        protected SqlConnection ObjConnection;
        protected SqlDataAdapter ObjAdapter;
        protected SqlParameter ObjParam;
        protected SqlCommand ObjCommand;
        protected SqlDataReader ObjReader;

        public void Dispose()
        {
            if (ObjConnection.State != ConnectionState.Closed)
            {
                ObjConnection.Close();
            }

            ObjConnection = null;
            ObjAdapter = null;
            ObjParam = null;
            ObjCommand = null;
            ObjReader = null;
        }

    }
}